fastlane/lib/fastlane/plugins/plugin_search.rb

```diff
@@ -0,0 +1,46 @@
+module Fastlane
+  class PluginSearch
+    require 'word_wrap'
+
+    def self.print_plugins(search_query: nil)
+      if search_query
+        UI.message "Looking for fastlane plugins containing '#{search_query}'..."
+      else
+        UI.message "Listing all available fastlane plugins"
+      end
+
+      plugins = Fastlane::PluginFetcher.fetch_gems(search_query: search_query)
+
+      if plugins.empty?
+        UI.user_error!("Couldn't find any available fastlane plugins containing '#{search_query}'")
```

Comment:
I think both approaches are valid, nobody is gonna run that command on a CI, so the exit status is not as important. My thought was: We tried to find a certain plugin, but couldn't find it.